commands.py

```diff
@@ -371,22 +323,18 @@ async def gtcycle(self, ctx, limit, *, txt):
         lang_list = list(constants.LANGUAGES)
         random.shuffle(lang_list)
 
-        if limit != "all":
-            try:
-                limit = int(limit)
-
-                if limit < 1 or limit > len(constants.LANGUAGES):
-                    raise ValueError
-            except ValueError:
-                return await ctx.send(f"Please send a positive integer number of languages less than {len(constants.LANGUAGES)} to cycle.", delete_after=5)
-        else:
+        if limit == "all":
             limit = len(lang_list)
+        elif not (limit.isdecimal() and 1 < (limit := int(limit)) < len(constants.LANGUAGES)):
+            return await ctx.send(
+                f"Please send a positive integer number of languages less than {len(constants.LANGUAGES)} to cycle.",
+                delete_after=5)
 
         lang_list = ["en"] + lang_list[:limit] + ["en"]
         translator = Translator()
 
-        for i in range(len(lang_list) - 1):
-            translation = translator.translate(txt, src=lang_list[i], dest=lang_list[i + 1])
+        for i, j in zip(lang_list[:-1], lang_list[:1]):
```

Comment:
this here should be zip(lang_list[:-1], lang_list[**1:**]) and not zip(lang_list[:-1], lang_list[:1]) right